core/src/main/java/com/linecorp/armeria/internal/common/RequestContextUtil.java

```diff
@@ -48,13 +55,45 @@
     private static final Set<Thread> REPORTED_THREADS =
             Collections.newSetFromMap(new MapMaker().weakKeys().makeMap());
 
+    private static final ContextStorage contextStorage;
+
+    static {
+        final List<ContextStorageProvider> providers = Streams.stream(
```

Comment:
ImmutableList.copyOf

Reply:
Isn't it okay since it calls `.collect(toImmutableList());`?

Reply:
Yeah but less code / complexity :)